internal/slicer/report.go

```diff
@@ -0,0 +1,47 @@
+package slicer
+
+import (
+	"io/fs"
+
+	"github.com/canonical/chisel/internal/fsutil"
+	"github.com/canonical/chisel/internal/setup"
+)
+
+type ReportEntry struct {
+	Path   string
+	Mode   fs.FileMode
+	Hash   string
+	Size   int
+	Slices []*setup.Slice
+	Link   string
+}
+
+// Report holds the information about files and directories created when slicing
+// packages.
+type Report struct {
+	Root string
+	// map indexed by path.
+	Entries map[string]ReportEntry
+}
+
+func NewReport(root string) *Report {
+	return &Report{Entries: make(map[string]ReportEntry), Root: root}
+}
+
+func (r *Report) AddEntry(slice *setup.Slice, entry fsutil.Info) {
+	if info, ok := r.Entries[entry.Path]; ok {
+		// Note: we do not check for clashes with the same path. That is done
+		// when parsing the slice definitions files and checking for conflicts.
```

Comment:
This is a Report type, which needs to be sensible by itself.  Documenting and expecting code living elsewhere to do something that unless it's done creates unexpected outcomes is an anti-pattern.

Reply:
I can remove the comment. What I am saying is that the report assumes that the slices and their paths are valid. I do not want to essentially duplicate the validation logic here and to have to keep both in sync.